cpp/include/cugraph/matrix_partition_device_view.cuh

```diff
@@ -148,6 +155,34 @@ class matrix_partition_device_view_t<vertex_t,
     return major_first_ + major_offset;
   }
 
+  // major_hypersparse_idx: index within the hypersparse segment
+  __host__ __device__ thrust::optional<vertex_t> get_major_hypersparse_idx_from_major_nocheck(
+    vertex_t major) const noexcept
+  {
+    if (dcs_nzd_vertices_) {
+      // we can avoid binary search (and potentially improve performance) if we add an auxiliary
+      // array or cuco::static_map (at the expense of additional memory)
```

Comment:
Possibly (at the expense of additional memory) for certain ranges of (# non-zero degree vertices) / (# vertices in the hypersparse segment).

Binary search based approach will be most memory efficient, and bloom-filter + binary search or hash-table based approaches may be faster for certain ranges of fill ratio (but will require more memory). We need to investigate the trade-offs (how much speedup we can get considering that this will require additional memory) once this turned out to be a performance bottleneck (it seems like it is not for most datasets we have tested and with a relatively small number of GPUs, but this may change for very low average vertex degree graphs and many GPUs... we need to implement few more optimizations to test for this setting).